tests/APIv2/ConversationsTest.php

```diff
@@ -0,0 +1,199 @@
+     *
+     * @return array Returns the fetched data.
+     */
+    public function testIndex() {
+        $nbsInsert = 3;
+
+        // Insert a few rows.
+        $rows = [];
+        for ($i = 0; $i < $nbsInsert; $i++) {
+            $rows[] = $this->testPost();
+        }
+
+        $result = $this->api()->get($this->baseUrl);
+        $this->assertEquals(200, $result->getStatusCode());
+
+        $rows = $result->getBody();
+        $this->assertGreaterThan($nbsInsert, count($rows));
+        // The index should be a proper indexed array.
+        for ($i = 0; $i < count($rows); $i++) {
+            $this->assertArrayHasKey($i, $rows);
+        }
+    }
+
+
+    /**
+     * Test DELETE /conversations/<id>/leave.
+     *
+     * @return array Returns the fetched data.
+     */
+    public function testDeleteLeave() {
+        $conversation = $this->testPost();
+
+        $result = $this->api()->delete(
+            "{$this->baseUrl}/{$conversation[$this->pk]}/leave"
+        );
+
+        $this->assertEquals(204, $result->getStatusCode());
+
+        $participantsResult = $this->api()->get(
+            "{$this->baseUrl}/{$conversation[$this->pk]}/participants"
+        );
+
+        $this->assertEquals(200, $participantsResult->getStatusCode());
+
+        $expectedFirstParticipant = [
+            'userID' => $this->api()->getUserID(),
+            'deleted' => true,
+        ];
+        $participants = $participantsResult->getBody()['participants'];
+
+        $this->assertTrue(is_array($participants));
+        $this->assertRowsEqual($expectedFirstParticipant, $participants[0]);
+    }
+
+    /**
+     * Test POST /conversations.
+     *
+     * @return array The conversation.
+     */
+    public function testPost() {
+        $postData = [
+            'participantIDs' => array_slice(self::$userIDs, 0, 2)
+        ];
+        $expectedResult = [
+            'insertUserID' => $this->api()->getUserID(),
+            'countParticipants' => 3,
+            'countMessages' => 0,
+            'countReadMessages' => 0,
+            'dateLastViewed' => null,
+        ];
+
+        $result = $this->api()->post(
+            $this->baseUrl,
+            $postData
+        );
+
+        $this->assertEquals(201, $result->getStatusCode());
+
+        $body = $result->getBody();
+        $this->assertTrue(is_int($body[$this->pk]));
+        $this->assertTrue($body[$this->pk] > 0);
+
+        $this->assertRowsEqual($expectedResult, $body, true);
+
+        return $body;
+    }
+
+    /**
+     * Test POST /conversations/<id>/participants.
+     */
+    public function testPostParticipants() {
+        $conversation = $this->testPost();
+
+        $postData = [
+            'participantIDs' => array_slice(self::$userIDs, 2)
+        ];
+        $result = $this->api()->post(
+            "{$this->baseUrl}/{$conversation[$this->pk]}/participants",
+            $postData
+        );
+
+        $this->assertEquals(201, $result->getStatusCode());
```

Comment:
I think you should go beyond just making sure a 201 is returned. Maybe perform a new GET on the conversation and verify the users were added. You can also verify/use the updated `countParticipants`.

Reply:
Following the recommendation that this should return something. I will check that the count participant was updated.

Reply:
Done.